src/wamr/wamr.cc

```diff
@@ -132,9 +133,41 @@ bool Wamr::load(std::string_view bytecode, std::string_view /*precompiled*/,
     return false;
   }
 
+  shared_module_ = wasm_module_share(module_.get());
+  if (shared_module_ == nullptr) {
+    return false;
+  }
+
   return true;
 }
 
+std::unique_ptr<WasmVm> Wamr::clone() {
+  assert(module_ != nullptr);
+
+  auto vm = std::make_unique<Wamr>();
+  if (vm == nullptr) {
+    return nullptr;
+  }
+
+  vm->store_ = wasm_store_new(engine());
+  if (vm->store_ == nullptr) {
+    return nullptr;
+  }
+
+  vm->module_ = wasm_module_obtain(vm->store_.get(), shared_module_.get());
+  if (vm->module_ == nullptr) {
+    return nullptr;
+  }
```

Comment:
Related to earlier question: is it a problem that `vm` here will be created with a null `shared_module_` pointer? It seems like if I clone `Wamr` instance A to get instance B, and then call `clone()` on B, that might not work since B's `shared_module_` isn't set?

Reply:
We only clone from the main thread to worker threads, not further, so while I get what you're saying, that's not an issue for the existing consumers.

Reply:
WAMR is in a `Cloneable::CompiledBytecode` level. Skip `load()` by using `clone()`. 

The `baseVM` always runs `load()` to initialize `share_module_`. All cloned `WasmVM` will call `clone()` directly.

Reply:
SG, thanks